ext/curl/tests/curl_setopt_CURLOPT_PREREQFUNCTION.phpt

```diff
@@ -0,0 +1,152 @@
+--TEST--
+Curl option CURLOPT_PREREQFUNCTION
+--EXTENSIONS--
+curl
+filter
+--SKIPIF--
+<?php
+$curl_version = curl_version();
+if ($curl_version['version_number'] < 0x075000) die("skip: test works only with curl >= 7.80.0");
+?>
+--FILE--
+<?php
+include 'server.inc';
+
+$host = curl_cli_server_start();
+$port = (int) (explode(':', $host))[1];
+
+$ch = curl_init();
+curl_setopt($ch, CURLOPT_URL, "{$host}/get.inc?test=file");
+curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+
+$result = curl_exec($ch);
+
+var_dump(CURLOPT_PREREQFUNCTION);
+var_dump(CURL_PREREQFUNC_OK);
+var_dump(CURL_PREREQFUNC_ABORT);
+
+$returnValue = CURL_PREREQFUNC_ABORT;
+
+echo "\nTesting with CURL_PREREQFUNC_ABORT\n";
+$callback = function() use ($port, &$returnValue) {
+	var_dump('callback');
+	var_dump(func_num_args());
+	$args = func_get_args();
+	var_dump(get_class($args[0]));
+	var_dump(filter_var($args[1], FILTER_VALIDATE_IP) !== false);
+	var_dump(filter_var($args[2], FILTER_VALIDATE_IP) !== false);
```

Comment:
Would it be possible to have the test not rely on the filter extension? Found this out while trying it locally without ext/filter compiled.

Otherwise it needs to be added to the EXTENSIONS section as I did.